src/Interpreters/Aggregator.cpp

```diff
@@ -853,7 +863,7 @@ void Aggregator::writeToTemporaryFile(AggregatedDataVariants & data_variants, co
 }
 void Aggregator::writeToTemporaryFile(AggregatedDataVariants & data_variants)
 {
-    return writeToTemporaryFile(data_variants, params.tmp_volume->getNextDisk()->getPath());
+    return writeToTemporaryFile(data_variants, params.tmp_volume->getDisk()->getPath());
```

Comment:
Why we have different logic here and https://github.com/ClickHouse/ClickHouse/pull/12670/files#diff-bafb5527e76244c7f83100af0572d862R771 here?

Reply:
Looks like a possible bug. @excitoon let's try copy logic from the [link](https://github.com/ClickHouse/ClickHouse/pull/12670/files#diff-bafb5527e76244c7f83100af0572d862R771) and check integration tests (I remember this was part of a test fix)

Reply:
Good catch!

Reply:
There is no such thing as `getNextDisk()` in `IVolume`. What shall we do? @NanoBjorn

Reply:
I did as you advised, but that looks ugly. Can we use some of polymorphism here?

Reply:
> I did as you advised, but that looks ugly. Can we use some of polymorphism here?

I thought about it for a long time, this seems the best way, despite the fact it is a bit ugly. So leave it this way:)